ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/KeyValueSubmission.scala

```diff
@@ -44,14 +46,24 @@ class KeyValueSubmission(metrics: Metrics) {
       tx: SubmittedTransaction,
   ): DamlSubmission =
     metrics.daml.kvutils.submission.conversion.transactionToSubmission.time { () =>
-      val inputDamlStateFromTx = InputsAndEffects.computeInputs(tx, meta)
       val encodedSubInfo = buildSubmitterInfo(submitterInfo)
+      val packageIdStates = meta.optUsedPackages
+        .getOrElse(
+          throw new InternalError("Transaction was not annotated with used packages")
+        )
+        .map(Conversions.packageStateKey)
+      val partyStates =
+        (tx.informees ++ submitterInfo.actAs).toList.map(Conversions.partyStateKey)
```

Comment:
How about encapsulating `(tx.informees ++ submitterInfo.actAs)` too somehow/somewhere?

Reply:
5f95fd0b3abaf0e7715307993edd36d476ff6e8f if you have a better name for it, I’m happy to change it. I considered `partyStates` but that doesn’t seem much better.